django/db/models/base.py

```diff
@@ -1741,6 +1753,49 @@ def _check_default_pk(cls):
             ]
         return []
 
+    @classmethod
+    def _check_composite_pk(cls):
+        errors = []
+        meta = cls._meta
+        pk = meta.pk
+
+        if not isinstance(pk, CompositePrimaryKey):
+            return errors
+
+        seen_columns = {}
+
+        for field_name in pk.field_names:
+            hint = None
+
+            try:
+                field = meta.get_field(field_name)
+            except FieldDoesNotExist:
+                field = None
+
+            if not field:
+                hint = "'%s' is not a valid field." % (field_name,)
+            elif not field.column:
+                hint = "'%s' field has no column." % (field_name,)
+            elif field.null:
+                hint = "'%s' field may not set 'null=True'." % (field_name,)
+            else:
```

Comment:
I think we should also not allow `GeneratedField`s
Currently I can use this, makemigrations, migrate and then get an infinite recursion error when trying to create an object 
<details><summary>Error if useful</summary>

```
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query_utils.py", line 215, in __get__
    if instance.pk is None and self.field.generated:
       ^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/fields/composite.py", line 289, in __get__
    return tuple(getattr(instance, attname) for attname in self.attnames)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/fields/composite.py", line 289, in <genexpr>
    return tuple(getattr(instance, attname) for attname in self.attnames)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query_utils.py", line 215, in __get__
    if instance.pk is None and self.field.generated:
       ^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/fields/composite.py", line 289, in __get__
    return tuple(getattr(instance, attname) for attname in self.attnames)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/fields/composite.py", line 289, in <genexpr>
    return tuple(getattr(instance, attname) for attname in self.attnames)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query_utils.py", line 215, in __get__
    if instance.pk is None and self.field.generated:
  File "/usr/local/lib/python3.11/site-packages/django/db/models/fields/composite.py", line 289, in <genexpr>
    return tuple(getattr(instance, attname) for attname in self.attnames)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query_utils.py", line 213, in __get__
    val = self._check_parent_chain(instance)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded
```
</details>
I don't see why anyone would want this so would be nice if it's just not allowed 👍

Reply:
You're right, I haven't thought about that.

Reply:
I pushed a fix for this.